src/testing/mock.rs

```diff
@@ -1,14 +1,17 @@
 use async_trait::async_trait;
 use codec::{Decode, Encode};
+use log::{debug, error};
+use parking_lot::Mutex;
+
+use tokio::task::yield_now;
```

Comment:
I think we are trying to avoid using `tokio`, so maybe use the equivalent from `futures`?

Reply:
I can't find anything similar in `futures`. There is something in `futures-lite`. Previous version was providing my own implementation, but the way you need to implement it when using `futures` looked awful.

Reply:
Right, there might be no equivalent in `futures`. :/ In this case I would still prefer eliminating `tokio` and keeping the previous version, even if it is a bit ugly. Not a strong preference though, so do what you will.